sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -1239,6 +1254,24 @@ object SQLContext {
     instantiatedContext.compareAndSet(null, sqlContext)
```

Comment:
@davies Seems when we create a sql context for a new session, instantiatedContext will be set to the context representing that session. Do you think if it makes sense to use instantiatedContext hold the root sql context (the one created directly from user-facing constructor instead of newSession)?